### DISCUSSION TITLE
Possible dakota.input.nspec discrepancy

### DISCUSSION THREAD

**Discussion by richardsonwk** [snl-dakota/dakota#65] [Q&A] [answered]
The Dakota 6.18 dakota.input.nspec file specifies a few list lengths via `LEN` that refer to other semi-unrelated optional keywords, which _seems_ wrong, but I don't know enough to know:

```
  [ discrete_design_set {0}
    ...
    [ real INTEGER > 0 {N_vam(sizet,numDiscreteDesSetRealVars)}
      ...
      [ categorical STRINGLIST LEN integer {N_vam(categorical,discreteDesignSetRealCat)}
    ...
  [ discrete_state_set {0}
    ...
    [ real INTEGER > 0 {N_vam(sizet,numDiscreteStateSetRealVars)}
      ...
      [ categorical STRINGLIST LEN integer {N_vam(categorical,discreteStateSetRealCat)} ]
      ...
  ( calibration_terms ALIAS least_squares_terms ALIAS num_least_squares_terms INTEGER >= 0 {N_rem(sizet,numLeastSqTerms)}
    ...
    [ field_calibration_terms INTEGER >= 0 {N_rem(sizet,numFieldLeastSqTerms)}
      ...
     ]
    ...
    [ 
      ( calibration_data {N_rem(true,calibrationDataFlag)}
        ...
        [ experiment_variance_type ALIAS variance_type STRINGLIST LEN1 field_calibration_terms {N_rem(strL,varianceType)} ]
```

**Comment by jadamstephens** [comment, marked as answer]
You are right. But, we're currently making only very limited use of the constraints that are encoded in the input spec, so these discrepancies aren't actively harming anything.

Were you hoping to use them for something?

**Reply by richardsonwk** [reply]
Our intended use is providing hints while a user is creating a Dakota configuration, e.g. something like "This list should have 4 elements but only has 3". Not at all important to fix! Thanks for taking the time to look.

**Reply by jadamstephens** [reply]
Our GUI does some of that, but support for it is limited. Because of your comment I talked to our GUI developer about it and discovered that the "LEN" expressions in particular aren't being evaluated, or at least not evaluated correctly. The CLI ignores the constraints entirely and relies on hard coding to do the same job.
